### DISCUSSION TITLE
How to use Next.js with Typescript, Jest, and baseUrl

### DISCUSSION THREAD

**Discussion by brno32** [vercel/next.js#22586] [Help]
I have the following `tsconfig.json`

```json
{
  "compilerOptions": {
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "baseUrl": ".",
    "jsx": "preserve",
    "lib": ["dom", "es2017"],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "preserveConstEnums": true,
    "removeComments": false,
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "target": "esnext",
    "forceConsistentCasingInFileNames": true,
    "esModuleInterop": true,
    "resolveJsonModule": true,
    "isolatedModules": true
  },
  "exclude": ["node_modules", ".next", "out"],
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"]
}
```

The only thing custom I've added is `"baseUrl": ".",` so that I can use absolute imports in my codebase. This works great, but Jest seems unable to find modules during tests

`Cannot find module 'src/path/to/module' from 'src/path/to/other/module'`

I've followed the configuration from https://github.com/vercel/next.js/tree/master/examples/with-typescript-eslint-jest, with the only difference I noticed being the `baseUrl` I've set in the `tsconfig.json`

**Comment by mithi** [comment]
Did you try putting the following in your `jest.config.js` ?
```
module.exports = {
    moduleDirectories: [".", "node_modules"],
}
```
